core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -332,6 +339,87 @@ private RexNode simplifyGenericNode(RexCall e) {
     return rexBuilder.makeCall(e.getType(), e.getOperator(), operands);
   }
 
+  /**
+   * Try to find a literal with the given value in the input list.
+   */
+  private int findLiteralIndex(List<RexNode> operands, long value) {
+    for (int i = 0; i < operands.size(); i++) {
+      if (operands.get(i).isA(SqlKind.LITERAL)) {
+        Comparable comparable = ((RexLiteral) operands.get(i)).getValue();
+        if (comparable instanceof BigDecimal && ((BigDecimal) comparable).longValue() == value) {
+          return i;
+        }
```

Comment:
Could you mark it that it only supports numeric type?

Reply:
Sure. I've made it explicit in the JavaDoc.